bc-backend/src/main/java/org/jboss/da/bc/impl/BuildConfigurationGeneratorImpl.java

```diff
@@ -92,16 +96,22 @@ private void validate(ProjectHiearchy hiearchy) throws IllegalStateException {
 
         ProjectDetail project = hiearchy.getProject();
 
-        if (project.isUseExistingBc() && !project.isBcExists())
+        if (project.isUseExistingBc() && !project.isBcExists()) {
+            project.addError(BcError.NO_EXISTING_BC);
             throw new IllegalStateException(
                     "Use existing build configuration is checked, but apperently there is not existing build configuration for "
                             + project.getGav());
+        }
 
-        if (!project.isUseExistingBc() && project.getEnvironmentId() == null)
+        if (!project.isUseExistingBc() && project.getEnvironmentId() == null) {
+            project.addError(BcError.NO_ENV_SELECTED);
             throw new IllegalStateException("Environment id is null for " + project.getGav());
+        }
 
-        if (!project.isUseExistingBc() && project.getProjectId() == null)
+        if (!project.isUseExistingBc() && project.getProjectId() == null) {
+            project.addError(BcError.NO_PROJECT_SELECTED);
             throw new IllegalStateException("Project id is null for " + project.getGav());
+        }
```

Comment:
I think it's pointless to add sth to project, when an exception is thrown. The project is discarded.